funcx_sdk/funcx/serialize/facade.py

```diff
@@ -25,9 +45,57 @@ def __init__(self):
         for key in METHODS_MAP_DATA:
             self.methods_for_data[key] = METHODS_MAP_DATA[key]()
 
+    def _start_off_process_checker(self):
+        """ Kicks off off_process_checker using subprocess and returns the port on which it
+        is listening.
+        """
+        std_file = f'/tmp/{uuid.uuid4()}'
+        logger.debug(f"Writing off_process_checker logs to: {std_file}")
+        std_out = open(std_file, 'w')
+        self.serialize_proc = subprocess.Popen(['off_process_checker.py'], stdout=std_out, stderr=subprocess.STDOUT)
+        with open(std_file) as f:
+            for i in range(200, 3000, 200):
```

Comment:
If any failure happens, this could block the `FuncXClient()` call for 2 + 4 + 6 + ... + 15 seconds, right? I wonder if there is a way to make it more transparent when it fails, or probably should at least let user know that we are doing something here? 

PS: I have not seen any failure on starting the checker server in my local test. So this is fine for me now. But just want to note here.

Reply:
We are a library, we can't print to screen. This rarely takes >1s. I'll add a broader `try` block and print out an exit message which can be used to terminate early in case of an obvious issue. Can you check the handling?

Reply:
Yes, this is a local server. If it does not work for ~5s, it is unlikely to work

Reply:
If there's a problem here, we exit early.